include/LightGBM/tree.h

```diff
@@ -147,6 +147,31 @@ class Tree {
 
   inline double split_gain(int split_idx) const { return split_gain_[split_idx]; }
 
+  inline double internal_value(int node_idx) const {
+    return internal_value_[node_idx];
+  }
+
+  inline bool IsNumericalSplit(int node_idx) const {
+    return !GetDecisionType(decision_type_[node_idx], kCategoricalMask);
+  }
+
+  inline int left_child(int node_idx) const { return left_child_[node_idx]; }
+
+  inline int right_child(int node_idx) const { return right_child_[node_idx]; }
+
+  inline int split_feature_inner(int node_idx) const {
+    return split_feature_inner_[node_idx];
+  }
+
+  inline int leaf_parent(int leaf_idx) const { return leaf_parent_[leaf_idx]; }
+
+  inline uint32_t threshold_in_bin(int node_idx) const {
+#ifdef DEBUG
+    CHECK(node_idx >= 0);
```

Comment:
why need this? could we avoid to pass that index?

Reply:
We don't need the debug flag, I will remove it.